tools/code-generation/smithy/codegen/cpp-smoke-tests/build.gradle.kts

```diff
@@ -56,44 +58,49 @@ tasks.register("generate-smithy-build") {
         val c2jMapStr: String = project.findProperty("c2jMap")?.toString() ?: ""
 
         fileTree(models).filter { it.isFile }.files.forEach eachFile@{ file ->
+
             val model = Model.assembler()
                     .addImport(file.absolutePath)
                     // Grab the result directly rather than worrying about checking for errors via unwrap.
                     // All we care about here is the service shape, any unchecked errors will be exposed
                     // as part of the actual build task done by the smithy gradle plugin.
                     .assemble().result.get();
-            val services = model.shapes(ServiceShape::class.javaObjectType).sorted().toList();
+
+            val services = model.shapes(ServiceShape::class.java).sorted().toList();
+
             if (services.size != 1) {
-                throw Exception("There must be exactly one service in each aws model file, but found " +
-                        "${services.size} in ${file.name}: ${services.map { it.id }}");
+                println("Unexpected number of service shapes detected: ${services.size} while processing file: ${file.name}. ")
```

Comment:
deliberately not throwing exception to allow other services to move on

Reply:
we should catch and throw issues when we find them, and leave them up to the caller, i.e. the CI script to ignore. for instance here, why are there more than 1 service shape? what is wrong with the smithy files that allows this to be the case?

Reply:
yes, but it affects all the other services that don't have issue. I can still throw it but it is going to show incomplete builds in pipeline if even one service has an issue

Reply:
the doesn't have anything to do with the pipeline, or CI at all, at pure level, if the code generator fails, it should bubble up the error to the caller, and the caller is responsible for handling it. in this case the CI pipeline.

> for instance here, why are there more than 1 service shape? what is wrong with the smithy files that allows this to be the case?

you didnt answer this question -- why do we need this and is this a data problem. what services cause this problem and why.